cli/admin/src/main/java/io/pravega/cli/admin/segmentstore/FlushToStorageCommand.java

```diff
@@ -46,35 +53,83 @@ public FlushToStorageCommand(CommandArgs args) {
-            int containerCount = getServiceConfig().getContainerCount();
-            for (int id = 0; id < containerCount; id++) {
-                flushContainerToStorage(adminSegmentHelper, id, segmentStoreHost);
-            }
+            startContainerId = 0;
+            endContainerId = getServiceConfig().getContainerCount() - 1;
         } else {
-            flushContainerToStorage(adminSegmentHelper, parseInt(containerId), segmentStoreHost);
+            startContainerId = parseInt(containerId);
+            endContainerId = getArgCount() == 2 ? parseInt(getArg(1)) : startContainerId;
+        }
+
+        for (int id = startContainerId; id <= endContainerId; id++) {
+            flushContainerToStorage(adminSegmentHelper, id);
         }
     }
 
-    private void flushContainerToStorage(AdminSegmentHelper adminSegmentHelper, int containerId, String segmentStoreHost) throws Exception {
+    private void flushContainerToStorage(AdminSegmentHelper adminSegmentHelper, int containerId) throws Exception {
         CompletableFuture<WireCommands.StorageFlushed> reply = adminSegmentHelper.flushToStorage(containerId,
-                new PravegaNodeUri(segmentStoreHost, getServiceConfig().getAdminGatewayPort()), super.authHelper.retrieveMasterToken());
+                new PravegaNodeUri(this.getHosts().get(containerId), getServiceConfig().getAdminGatewayPort()), super.authHelper.retrieveMasterToken());
         reply.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
         output("Flushed the Segment Container with containerId %d to Storage.", containerId);
     }
 
     public static CommandDescriptor descriptor() {
         return new CommandDescriptor(COMPONENT, "flush-to-storage", "Persist the given Segment Container into Storage.",
-                new ArgDescriptor("container-id", "The container Id of the Segment Container that needs to be persisted, " +
-                        "if given as \"all\" all the containers will be persisted."),
-                new ArgDescriptor("segmentstore-endpoint", "Address of the Segment Store we want to send this request."));
+                new ArgDescriptor("start-container-id", "The start container Id of the Segment Container that needs to be persisted, " +
+                        "if given as \"all\" all the containers will be persisted."), new ArgDescriptor("end-container-id", "The end container Id of the Segment Container that needs to be persisted, " +
+                "if given as \"all\" all the containers will be persisted."));
+    }
+
+    private Map<Integer, String> getHosts() {
+        Map<Host, Set<Integer>> hostMap;
+        try {
+            @Cleanup
+            ZKHelper zkStoreHelper = ZKHelper.create(getServiceConfig().getZkURL(), getServiceConfig().getClusterName());
+            hostMap = zkStoreHelper.getCurrentHostMap();
+        } catch (Exception e) {
+            System.err.println("Exception accessing to Zookeeper cluster metadata: " + e.getMessage());
```

Comment:
We do not keep any system.err/out .  Maybe you can add the message to the exception thrown below and remove this line or use a "output" if you have to convey something to the user.

Reply:
Removed that line